### PR TITLE
refactor: Use Api streamwrapper

### PR DESCRIPTION
**Pull request by dbrasseur-aneo**

### CODE REVIEW

File: Client/src/Common/Submitter/BaseClientSubmitter.cs [hunk 1]
```diff
@@ -223,105 +224,39 @@ public Output GetTaskOutputInfo(string taskId)
...
-        asyncClientStreamingCall.RequestStream.CompleteAsync()
-                                .Wait();
-
-        var createTaskReply = asyncClientStreamingCall.ResponseAsync.Result;
-
-        switch (createTaskReply.ResponseCase)
-        {
-          case CreateTaskReply.ResponseOneofCase.None:
-            throw new Exception("Issue with Server !");
-          case CreateTaskReply.ResponseOneofCase.CreationStatusList:
```
Comment (aneojgurhem) [inline Client/src/Common/Submitter/BaseClientSubmitter.cs:318]: Did we move this error management in Api ?

File: Client/src/Common/Submitter/BaseClientSubmitter.cs [hunk 2]
```diff
@@ -223,105 +224,39 @@ public Output GetTaskOutputInfo(string taskId)
...
+                                                   {
+                                                     var taskRequest = new TaskRequest
+                                                                       {
+                                                                         Payload = UnsafeByteOperations.UnsafeWrap(pwd.Item2),
+                                                                       };
+                                                     taskRequest.DataDependencies.AddRange(pwd.Item3 ?? Enumerable.Empty<string>());
+                                                     taskRequest.ExpectedOutputKeys.Add(pwd.Item1);
+                                                     return taskRequest;
+                                                   })
+                                           .ToList();
```
Comment (aneojgurhem) [inline Client/src/Common/Submitter/BaseClientSubmitter.cs:237]: We wanted to avoid this `ToList()` with this implementation

Comment (lemaitre-aneo) [reply Client/src/Common/Submitter/BaseClientSubmitter.cs:237]: With the retry, either you materialize the `requests` list, or you iterate the `payloadsWithDependencies` multiple times.

Comment (dbrasseur-aneo) [reply Client/src/Common/Submitter/BaseClientSubmitter.cs:237]: I don't mind either way, I can put it directly but we risk multiple enumeration. Your call

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**